sklearn/utils/estimator_checks.py

```diff
@@ -3147,3 +3155,15 @@ def check_n_features_in_after_fitting(name, estimator_orig):
 
     with raises(ValueError, match=msg):
         estimator.partial_fit(X_bad, y)
+
+
+def check_estimator_get_tags_default_keys(name, estimator_orig):
+    # check that if _get_tags is implemented, it contains all keys from
+    # _DEFAULT_KEYS
+    estimator = clone(estimator_orig)
+    if not hasattr(estimator, "_get_tags"):
+        return
+
+    tags_keys = set(estimator._get_tags().keys())
+    default_tags_keys = set(_DEFAULT_TAGS.keys())
+    assert tags_keys.intersection(default_tags_keys) == default_tags_keys
```

Comment:
We need a new test in `test_estimator_checks.py ` for this check, no?

Reply:
Also shouldn't raise a more descriptive error message such as `f"{estimator.__name___}._get_tags is missing entries for the following default tags: {missing_tag_keys}"` where:

```python
missing_tag_keys = default_tags_keys - tags_keys.intersection(default_tags_keys)
```
(assuming I am not mistaken in my set operator logic ;)